src/main/java/in/projecteka/consentmanager/clients/ClientRegistryClient.java

```diff
@@ -62,15 +62,42 @@ public ClientRegistryClient(WebClient.Builder webClientBuilder, String baseUrl)
                 .bodyToMono(Session.class);
     }
 
+    public Mono<Session> getTokenWithRefreshToken(String clientId, String clientSecret, String refreshToken) {
+        return webClientBuilder.build()
+                .post()
+                .uri("/api/1.0/sessionsRefresh")
+                .contentType(APPLICATION_JSON)
+                .accept(APPLICATION_JSON)
+                .body(BodyInserters.fromValue(requestWithRefreshToken(clientId, clientSecret, refreshToken)))
+                .retrieve()
+                .onStatus(HttpStatus::isError, clientResponse -> clientResponse.bodyToMono(Properties.class)
+                        .doOnNext(properties -> logger.error(properties.toString()))
+                        .thenReturn(ClientError.unAuthorized()))
```

Comment:
Doesn't seem to be correct to consider all the error as unauthorized error. If the exception is internally handled (I guess so), I would say, instead of saying unauthorized, return something went wrong since it was supposed to work always.

Reply:
It might have been followed for sometime, but when you are working on something, don't follow blindly, if you think that does not make sense, question it,  discuss it and correct it (if everyone agreed :) )